### PR TITLE
refactor: login page and validation

### PR DESCRIPTION
**Pull request by jose-sal-y-rosas**
## Issue ticket number and link

https://github.com/CHUPITO-Org/FE-Chupito/issues/46

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [ ] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [x] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

- Login component has redirect validation

## What is the new behavior?

- Moved all validations for redirect to Login Page

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by anversa-pro — APPROVED** [review]
✅

File: src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -46,19 +51,38 @@ export default function LoginPage(): JSX.Element {
...
       }
     } catch (error) {
       console.error('Google login error:', error)
     } finally {
       setLoading(false)
     }
   }
+
+  const handleRedirect = useCallback(() => {
+    const userState = user.state
```
Comment (jlora-mck) [inline src/pages/Login/Login.tsx:63]: Can we try this ?
```suggestion
    const { username, isAdmin } = user.state
```

File: src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -46,19 +51,38 @@ export default function LoginPage(): JSX.Element {
...
+
+    let shouldRedirectTo = '/login'
+
+    if (userState.username) {
+      shouldRedirectTo = userState.isAdmin ? '/events/list' : '/'
+    } else if (eventId) {
+      shouldRedirectTo = `/event-info/${eventId}`
+    }
+    history.push(shouldRedirectTo)
+  }, [eventId, history, user.state])
```
Comment (jlora-mck) [inline src/pages/Login/Login.tsx:73]: i dont think we need to add user.state to deps

Comment (jose-sal-y-rosas) [reply src/pages/Login/Login.tsx:73]: I I think If I change this, it could break the application because there is not test

**Review by jlora-mck — APPROVED** [review]

**Pull request merged**